agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/classicsdk/BytecodeUtilImpl.java

```diff
@@ -67,10 +69,15 @@ public void setConnectionString(String connectionString) {
               + " \"connectionStringConfiguredAtRuntime\" to true");
       return;
     }
-    if (TelemetryClient.getActive().getConnectionString() != null) {
-      logger.warn("Connection string is already set");
-      return;
+
+    if (!connectionStringProgrammaticallySet.getAndSet(true)
+        && TelemetryClient.getActive().getConnectionString() != null) {
+      logger.info(
+          "The connection string is programmatically set. It will take precedence over the value defined from the applicationinsights.json file or the "
+              + APPLICATIONINSIGHTS_CONNECTION_STRING_ENV
+              + " environment variable.");
```

Comment:
can use placeholder `{}`

Reply:
Not sure to follow because no value to change